packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -979,7 +979,8 @@ export class MergeTree {
 					}
 					if (
 						seqLTE(removalInfo.removedSeq, refSeq) ||
-						removalInfo.removedClientIds.includes(clientId)
+						(removalInfo.removedClientIds !== undefined &&
```

Comment:
Suggest using non-null assertion here too instead of the extra undefined check to avoid any runtime changes.